### ISSUE TITLE
[bug] "Block header bits do not match target" error on blocks 94751, 481824 and 709632

### ISSUE THREAD

**Issue by m-kus** [keep-starknet-strange/raito#137]
How to reproduce:
```
scarb run integration_tests tests/data/light_481823.json
scarb run integration_tests tests/data/light_709631.json
```

How to troubleshoot:
1. Check if the calculated difficulty target matches the expected one (see `expected` field in the test data json)
2. If it's not the case then review the target calculation algorithm

See also:
- https://learnmeabitcoin.com/technical/block/bits/
- https://github.com/bitcoin/bitcoin/blob/f794a0d5f4041eb8beaff7a6f7f2867d9507dc58/src/pow.cpp

When the bug is fixed:
- Add unit test that checks for that specific edge case
- Remove failing integration test from ignored in the "integration_tests.sh"

Suggested deadline: 9 Sep

**Comment by raizo07** [comment]
@m-kus Can I work on this?

**Comment by Jeanmichel7** [comment]
My PR #155 also seems to solve this problem

**Comment by m-kus** [comment]
Fixed by #160
